File: bespin/src/main/java/org/example/product/bespin/BespinModule.java
package org.example.product.bespin;

import org.example.product.corellia.CorelliaModule;
import org.example.product.coruscant.CoruscantModule;
import org.apache.velocity.io.VelocityWriter;

import java.io.FileWriter;
import java.io.IOException;

public class BespinModule {

    public CorelliaModule doThings() throws IOException {
        new CoruscantModule();
        new VelocityWriter(new FileWriter(System.getProperty("java.io.tmpdir") + "/dummy.out"));
        return new CorelliaModule();
    }
}

File: bespin/src/main/kotlin/org/example/product/bespin/BespinModule.kt
package org.example.product.bespin

import org.apache.velocity.io.VelocityWriter
import org.example.product.corellia.CorelliaModule
import org.example.product.coruscant.CoruscantModule
import java.io.FileWriter

class BespinModule {

    fun doThings(): CorelliaModule {
        CoruscantModule()
        VelocityWriter(FileWriter(System.getProperty("java.io.tmpdir") + "/dummy.out"))
        return CorelliaModule()
    }
}